kodilla-patterns/src/main/java/com/kodilla/patterns/prototype/library/Library.java

```diff
@@ -0,0 +1,41 @@
+package com.kodilla.patterns.prototype.library;
+
+import com.kodilla.patterns.prototype.Prototype;
+import java.util.HashSet;
+import java.util.Set;
+
+public final class Library extends Prototype {
+    private String name;
+    private Set<Book> books = new HashSet<>();
+
+    public Library(final String name) {
+        this.name = name;
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public Set<Book> getBooks() {
+        return books;
+    }
+
+    @Override
+    public String toString() {
+        return "books=" + books;
+    }
+
+    public Library shallowCopy() throws CloneNotSupportedException {
+        return (Library) super.clone();
+    }
+
+    public Library deepCopy() throws CloneNotSupportedException {
+        Library clonedBoard = (Library) super.clone();
+        clonedBoard.books = new HashSet<>();
+        for( Book book : books) {
+            Book clonedList = new Book(book.getTitle(), book.getAuthor(), book.getPublicationDate());
```

Comment:
chyba clonedList nie jest najszczęśliwszą nazwą wybraną dla obiektu typy Book, prawda? :-) Sugeruję po prostu book lub clonedBook. Ze względu na to, że klasa Book jest immutable to tak naprawdę nie trzeba clonować poszczególnych elementów book, ale Twoje rozwiązanie jest lepsze i bezpieczniejsze nawet gdy Book przestanie być immutable.